### DISCUSSION TITLE
Progressive habits

### DISCUSSION THREAD

**Discussion by RedApCi** [iSoron/uhabits#839] [Feature Requests] [answered]
would be cool if there were a way to set kinda progressive habits that level up in difficulty. Say you want to read more so you set a habit of reading for 30 mins a day. Once you've locked in this habit you might want to increase it to 45 mins. would be cool if you could use the same habit instead of creating a new one and just increase the difficulty and have this somehow reflected in the score.

**Comment by iSoron** [comment, marked as answer]
Thank you for the feature request. This seems related to #270.

**Comment by hiqua** [comment]
I think it might be enough to just record the current habit fields along with the checkmarks, and adapt the same habit according to your goals.

Record the date of changes, so that checkmarks recorded in-between are associated with the previous state. Maybe allow the user to record manually that the goal has changed (rather than just a typo fix), to avoid possible noise.

So if you changed the habit on 2021-01-01, the checkmarks in 2021 are associated with 45min, the ones before with 30min.

I think what you're hinting at is rather an automation of this increase in goal, e.g. specify that the goal is numeric, and IFF streak is longer than $DAYS, then increase goal by $STEP.

**Reply by RedApCi** [reply]
Good idea! Being able to automate the increase would be even better. Although then I guess you would have to be able to set some sort of ceiling (unless you really want to be reading 24 hours a day?!). 
If numeric goals were possible you could also have that included in the scoring. i.e. if the goal is 30mins the app would not treat 5mins and 29mins the same.

**Reply by hiqua** [reply]
The numerical goals are already implemented and will come with the v2, we can revisit then!

**Reply by RedApCi** [reply]
> The numerical goals are already implemented and will come with the v2, we can revisit then!

Awesome!

**Comment by atkirtland** [comment]
Since numerical goals are in, maybe this could be revisited? I would find it really useful to be able to record different goals for each day/week as some goals, like weekly running distance, often vary over time.

**Comment by SzczurekYT** [comment]
Yes please, I'm very much surprised such a basic feature has not been implemented over the years. Especially for numeric habits,  for most of them the target goal value will increase or decrease as you get better, and I don't want to alter old history, and I don't want to have duplicate habits for one thing with different values.

For me a simple behavior of the app only applying the goal change forward, and keeping the history intact would be enough, no need for complicated editing ui or anything like that.

I'm pretty busy, but if could find some time I could probably even implement that myself, as lack of this is a pretty big issue for me, and so far stopped me from increasing a goal in the app, so for a long time now the app doesn't really reflect if I met my goal or not 😔

**Reply by sunkup** [reply]
Agree. Would be super nice to have an option near the numerical target that says: "increase target by amount X every (day/week/month/year).

So say, for example,  you do 30 reps of an exercise for a week and the next week it _automatically_ increases by 1 and your target becomes 31 reps for a week etc.
